### ISSUE TITLE
edgetpu_compiler support for TF2.x quantization and conversion 

### ISSUE THREAD

**Issue by john-maidbot** [google-coral/edgetpu#366]
tensorflow-gpu version: 2.4.1
Edgetpu compiler version: 15.0.340273435

Hello, I am trying to convert a TF 2.x model to tflite and compile it for the edgetpu. Specifically I would like to perform quantization aware training (QAT) to ensure my final model has as little accuracy decay as possible.

I saw on the Coral.ai website that you do not currently support QAT for TF 2.x, only post-training quantization:

"If you're using TensorFlow 2, the only way to quantize the input and output tensors is with post-training quantization in TensorFlow r2.3 or higher (see this tutorial). Currently, you cannot quantize the input/output tensors using quantization-aware training in TensorFlow 2."
source: https://coral.ai/docs/edgetpu/models-intro/#compiling

I do not know how old these docs are, but this seems to still be accurate as when I try to perform QAT on the simple mnist example from the TF website (https://www.tensorflow.org/model_optimization/guide/quantization/training_example), the edgetpu compiler says that the model is not quantized. Whereas if I perform post-training quantization, it works just fine. 

Does your group have plans on providing QAT support for TF 2.x in the near future? Or is there already QAT support for TF 2.x, and perhaps I am doing something wrong in my conversion code (below)?

Thanks for your time!

P.S. If this is a repeat of an existing issue, please point me to it. I could not find any open issues asking about this.

``` 
import tempfile
import os
import tensorflow as tf
from tensorflow import keras
import tensorflow_model_optimization as tfmot
import numpy as np

physical_devices = tf.config.list_physical_devices('GPU')
tf.config.experimental.set_memory_growth(physical_devices[0], True)

# Load MNIST dataset
mnist = keras.datasets.mnist
(train_images, train_labels), (test_images, test_labels) = mnist.load_data()

# Normalize the input image so that each pixel value is between 0 to 1.
train_images = train_images / 255.0
test_images = test_images / 255.0

# Define the model architecture.
model = keras.Sequential([
  keras.layers.InputLayer(input_shape=(28, 28)),
  keras.layers.Reshape(target_shape=(28, 28, 1)),
  keras.layers.Conv2D(filters=12, kernel_size=(3, 3), activation='relu'),
  keras.layers.MaxPooling2D(pool_size=(2, 2)),
  keras.layers.Flatten(),
  keras.layers.Dense(10)
])

# Train the digit classification model
model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

model.fit(
  train_images,
  train_labels,
  epochs=1,
  validation_split=0.1,
)

# q_aware stands for for quantization aware.
quantize_model = tfmot.quantization.keras.quantize_model
q_aware_model = quantize_model(model)

# `quantize_model` requires a recompile.
q_aware_model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

q_aware_model.summary()

# train test split
train_images_subset = train_images[0:1000] # out of 60000
train_labels_subset = train_labels[0:1000]

# do some quantization fine tuning
q_aware_model.fit(train_images_subset, train_labels_subset,
                  batch_size=500, epochs=1, validation_split=0.1)

_, baseline_model_accuracy = model.evaluate(
    test_images, test_labels, verbose=0)

_, q_aware_model_accuracy = q_aware_model.evaluate(
   test_images, test_labels, verbose=0)

q_aware_model.save('q_aware_model')

print('Baseline test accuracy:', baseline_model_accuracy)
print('Quant test accuracy:', q_aware_model_accuracy)

# # Quantization aware training - does not compile for edgetpu
# converter = tf.lite.TFLiteConverter.from_keras_model(q_aware_model)
# converter.optimizations = [tf.lite.Optimize.DEFAULT]
# quantized_tflite_model = converter.convert()

# Post training quantization - compiles for edgetpu
def generator():
    for image in train_images[:100]:
        yield [np.array([image.astype(np.float32)])]
converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.experimental_new_converter = False
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
converter.inference_input_type = tf.uint8
converter.inference_output_type = tf.uint8
converter.representative_dataset = tf.lite.RepresentativeDataset(generator)
quantized_tflite_model = converter.convert()


with open("test.tflite", "wb") as f:
    f.write(quantized_tflite_model)

# on command line, compile the tflite model
# edgetpu_compiler -sa test.tflite
```

**Comment by Naveen-Dodda** [comment]
Hello, 

While there are few models not fully supported on Coral with QAT , the has not been updated. But you can still few model with QAT to Coral platform. 

I see that your above example is inspired from [QAT example](https://www.tensorflow.org/model_optimization/guide/quantization/training_example)

I was able to compile the model by just fixing a input shape of the model. (this chnages shape form (None,28,28) to (1,28,28)

q_aware_model.input.set_shape((1,) + q_aware_model.input.shape[1:])
q_aware_model.input.shape

converter = tf.lite.TFLiteConverter.from_keras_model(q_aware_model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]

Optional
#converter.inference_input_type = tf.uint8
#converter.inference_output_type = tf.uint8
#converter.experimental_new_converter = True

quantized_tflite_model = converter.convert()
open("quantized_tflite_model.tflite", "wb").write(quantized_tflite_model)


now qunatized_tflite_model should be pass the edgetpu_compiler

**Comment by caijinana** [comment]
Have you slove the problem?

**Comment by caijinana** [comment]
@Naveen-Dodda i have a create a model with tf2.2, using QAT and got .tflite,but in some model which have BN , edgetpu_compiler shows model not quantization
